prometheusx/metrics.go

```diff
@@ -1,32 +1,69 @@
 package prometheus
 
 import (
+	"net/http"
+	"strconv"
+
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/urfave/negroni"
 )
 
 // Metrics prototypes
 type Metrics struct {
-	ResponseTime *prometheus.HistogramVec
+	responseTime    *prometheus.HistogramVec
+	totalRequests   *prometheus.CounterVec
+	duration        *prometheus.HistogramVec
+	responseSize    *prometheus.HistogramVec
+	requestSize     *prometheus.HistogramVec
+	handlerStatuses *prometheus.CounterVec
 }
 
 // Method for creation new custom Prometheus  metrics
 func NewMetrics(app, version, hash, date string) *Metrics {
+	labels := map[string]string{
+		"version":   version,
```

Comment:
Would it maybe make sense to add the `app` to the label instead of prefixing it?

Reply:
It would make sense, but I didn't want to break previous metrics names as someone could already use it

Reply:
Ok cool, then I propose to change it right away :)

Do you maybe have a doc explaining what some best practices are for these type of metrics?

Reply:
Sure, I will update it then. For the metrics, we wanna follow and deliver basic "RED" metrics, were we can found the Rate, Error and Duration. This is example article, but I'm sure that you could find more https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/.